README.md
![alt XOOPS CMS](https://xoops.org/images/logoXoops4GithubRepository.png)
## Xoosla Candy for [XOOPS CMS 2.5.8+](https://xoops.org)
[![XOOPS CMS Module](https://img.shields.io/badge/XOOPS%20CMS-Module-blue.svg)](https://xoops.org)
[![Software License](https://img.shields.io/badge/license-GPL-brightgreen.svg?style=flat)](http://www.gnu.org/licenses/gpl-2.0.html)

[![Scrutinizer Code Quality](https://img.shields.io/scrutinizer/g/mambax7/xooslacandy.svg?style=flat)](https://scrutinizer-ci.com/g/mambax7/xooslacandy/?branch=master)
[![Codacy Badge](https://api.codacy.com/project/badge/grade/2d27c0023ee54f0b9ba2b5d17a68b2a5)](https://www.codacy.com/app/mambax7/xooslacandy)
[![Code Climate](https://img.shields.io/codeclimate/github/mambax7/xooslacandy.svg?style=flat)](https://codeclimate.com/github/mambax7/xooslacandy)
[![SensioLabsInsight](https://insight.sensiolabs.com/projects/9dc918fe-ea63-4675-832c-8f6c74cdf78f/mini.png)](https://insight.sensiolabs.com/projects/9dc918fe-ea63-4675-832c-8f6c74cdf78f)
[![Latest Pre-Release](https://img.shields.io/github/tag/XoopsModules25x/xooslacandy.svg?style=flat)](https://github.com/XoopsModules25x/xooslacandy/tags/)
[![Latest Version](https://img.shields.io/github/release/XoopsModules25x/xooslacandy.svg?style=flat)](https://github.com/XoopsModules25x/xooslacandy/releases/)

**Xoosla Candy** for [XOOPS CMS](https://xoops.org) Xoosla Candy was inspired by the Joomla modules, RokCandy by Rocket themes. Xooslacandy works in very much the same way but for Xoops.
This module was basically just a test for the new Xoosla module framework.

[![Tutorial Available](https://xoops.org/images/tutorial-available-blue.svg)](https://www.gitbook.com/book/xoops/xooslacandy-tutorial/) Tutorial: see [GitBook](https://www.gitbook.com/book/xoops/xooslacandy-tutorial/). 
To contribute to the Tutorial, [fork it on GitHub](https://github.com/XoopsDocs/xooslacandy-tutorial)

[![Translations on Transifex](https://xoops.org/images/translations-transifex-blue.svg)](https://www.transifex.com/xoops) 

Please visit us on https://xoops.org

**Current and upcoming "next generation" versions of XOOPS CMS are being crafted on GitHub at: https://github.com/XOOPS**


docs/changelog.txt
<h5>1.00 Beta 1  [NOT RELEASED]</h5>  Dev: Xoops 2.5.9, PHP 7.2.3
<hr>
- XOOPS 2.5.9, PHP 7 (mamba)
- removed @version (mamba)
- PSR-2 code cosmetics (mamba)
- addNavigation(basename(__FILE__))  (mamba)
- dirname(__FILE__) to __DIR__ (mamba)
- replaced < br /> with < br > (mamba)
- Unnecessary double quotes (mamba)
- reference mismatch (mamba)
- Hardening to type safe (mamba)
- changed _handler to Handler (mamba)
- added 'true' for tokens in forms (mamba)
- renamed .html Smarty templates to .tpl (mamba)
- removed '/' from void HTML elements (&lt;br&gt;, &lt;img&gt;, etc) (mamba)
- HTML 5: removed 'selected' and 'checked' (mamba)
- Updates for PayPal changes (zyspec)
- added Contributing.md (mamba)


<h5>1.00 Alpha 1  [2010-07-05]</h5>
<hr>
- Initial release (Catzwolf)


xoops_version.php
<?php
/**
 * Name: Xoops Version
 * Description: Xoops Version file
 *
 * @package    : Xoosla Modules
 * @Module     : Xoosla Candy
 * @subpackage :
 * @since      : v1.00
 * @author     John Neill <catzwolf@xoosla.com>
 * @copyright  : Copyright (C) 2010 Xoosla Modules. All rights reserved.
 * @license    : GNU/LGPL, see docs/license.php
 */
defined('XOOPS_ROOT_PATH') || die('Restricted access');

require_once __DIR__ . '/preloads/autoloader.php';

/**
 * Module Details
 */

$modversion['version']             = 1.00;
$modversion['requires']            = 1.00;
$modversion['module_status']       = 'Beta 1';
$modversion['release_date']        = '2017/05/14';
$modversion['name']                = _XL_XCA_CANDY;
$modversion['description']         = _XL_XCA_CANDY_DSC;
$modversion['author']              = XL_MI_XOOSLA_AUTHOR;
$modversion['credits']             = XL_MI_XOOSLA_CREDITS;
$modversion['status']              = XL_MI_XOOSLA_STATUS;
$modversion['lead']                = XL_MI_XOOSLA_LEAD;
$modversion['contributors']        = XL_MI_XOOSLA_CONTRIBUTORS;
$modversion['website_url']         = 'http://www.xoosla.com';
$modversion['website_name']        = 'Xoosla Modules';
$modversion['email']               = 'support@xoosla.com';
$modversion['demo_site_url']       = '';
$modversion['demo_site_name']      = '';
$modversion['support_site_url']    = '';
$modversion['support_site_name']   = 'Offical Support Website';
$modversion['submit_bug_url']      = 'http://code.google.com/p/xoosla-modules/issues/list';
$modversion['submit_bug_name']     = 'Bug Submission';
$modversion['submit_feature_url']  = '';
$modversion['submit_feature_name'] = '';
$modversion['disclaimer']          = _XL_XCA_DISCLAIMER;
$modversion['license']             = 'GPL see LICENSE';
$modversion['official']            = 0;
$modversion['image']               = 'images/xooslacandy_mlogo.png';
$modversion['dirname']             = basename(__DIR__);

/**
 * SQL
 */
$modversion['sqlfile']['mysql'] = 'sql/mysql.sql';

/**
 * Mysql Tables
 */
$modversion['tables'][] = 'xca_candy';

/**
 * Admin things
 */
$modversion['hasAdmin']    = 1;
$modversion['system_menu'] = 1;
$modversion['adminindex']  = 'admin/index.php';
$modversion['adminmenu']   = 'admin/menu.php';

/**
 * Additionnal script executed during install update
 */
$modversion['onInstall']   = 'include/oninstall.php';
$modversion['onUpdate']    = 'include/onupdate.php';
$modversion['onUninstall'] = 'include/onuninstall.php';

/**
 * Frontend
 */
$modversion['hasMain'] = 0;

/**
 * Frontend
 */
$modversion['hasSearch'] = 0;

/**
 * Comments
 */
$modversion['hasComments'] = 0;

/**
 * Notifications
 */
$modversion['hasNotification'] = 0;

// ------------------- Help files ------------------- //
$modversion['helpsection'] = [
    ['name' => _MI_XOOSLA_OVERVIEW, 'link' => 'page=help'],
    ['name' => _MI_XOOSLA_DISCLAIMER, 'link' => 'page=disclaimer'],
    ['name' => _MI_XOOSLA_LICENSE, 'link' => 'page=license'],
    ['name' => _MI_XOOSLA_SUPPORT, 'link' => 'page=support'],
];

/**
 * Blocks
 */

/**
 * Templates
 */
#$modversion['templates'][] = array(
#    'file' => '',
#    'description' => ''
#    );
/**
 * Module Configuration
 */
$modversion['config'][] = [
    'name'        => 'usemacros',
    'title'       => '_XL_MI_ADM_USEMACROS',
    'description' => '_XL_MI_ADM_USEMACROS_DSC',
    'formtype'    => 'group_multi',
    'valuetype'   => 'array',
    'default'     => ['1 2 3']
];
